File: core/src/Dignite.Paperbase.Application.Contracts/Permissions/PaperbasePermissions.cs
using Volo.Abp.Reflection;

namespace Dignite.Paperbase.Permissions;

public class PaperbasePermissions
{
    public const string GroupName = "Paperbase";

    public static class Documents
    {
        public const string Default = GroupName + ".Documents";
        public const string Upload = Default + ".Upload";
        public const string Delete = Default + ".Delete";
        public const string PermanentDelete = Default + ".PermanentDelete";
        public const string Restore = Default + ".Restore";
        public const string Export = Default + ".Export";
        public const string ConfirmClassification = Default + ".ConfirmClassification";

        public static class Pipelines
        {
            public const string Default = Documents.Default + ".Pipelines";
            public const string Retry = Default + ".Retry";
        }
    }

    public static string[] GetAll()
    {
        return ReflectionHelper.GetPublicConstantsRecursively(typeof(PaperbasePermissions));
    }
}

File: angular/packages/paperbase/src/lib/shared/tokens/paperbase-permissions.ts
export const PAPERBASE_PERMISSIONS = {
  Documents: {
    Default: 'Paperbase.Documents',
    Upload: 'Paperbase.Documents.Upload',
    Delete: 'Paperbase.Documents.Delete',
    Restore: 'Paperbase.Documents.Restore',
    Export: 'Paperbase.Documents.Export',
    ConfirmClassification: 'Paperbase.Documents.ConfirmClassification',
    Chat: {
      Default: 'Paperbase.Documents.Chat',
      Create: 'Paperbase.Documents.Chat.Create',
      SendMessage: 'Paperbase.Documents.Chat.SendMessage',
      Delete: 'Paperbase.Documents.Chat.Delete',
    },
    Pipelines: {
      Default: 'Paperbase.Documents.Pipelines',
      Retry: 'Paperbase.Documents.Pipelines.Retry',
    },
  },
  DocumentRelations: {
    Default: 'Paperbase.DocumentRelations',
    Create: 'Paperbase.DocumentRelations.Create',
    Delete: 'Paperbase.DocumentRelations.Delete',
    ConfirmRelation: 'Paperbase.DocumentRelations.ConfirmRelation',
  },
} as const;
